src/read.jl

```diff
@@ -104,113 +104,155 @@ function read_storage(filepath)::Storage
-            if !demand_flexibility["enabled"]
-                demand_flexibility["interval_balance"] = false
-                demand_flexibility["rolling_balance"] = false
+    # Try loading the demand flexibility and demand flexibility cost profiles
+    for s in ["up", "dn"]
+        # Pre-specify the demand flexibility and demand flexibility cost profiles
+        demand_flexibility["flex_amt_" * s] = nothing
+        demand_flexibility["cost_" * s] = nothing
+
+        # Only try loading the profiles if demand flexibility is enabled
+        if demand_flexibility["enabled"] == "not_specified" || (
+            demand_flexibility["enabled"]
+        )
+            # Try loading the demand flexibility profiles
+            try
+                demand_flexibility["flex_amt_" * s] = CSV.File(
+                    joinpath(filepath, "demand_flexibility_" * s * ".csv")
+                ) |> DataFrames.DataFrame
+                println("...loading demand flexibility " * s * " profiles")
+            catch e
+                println("Demand flexibility " * s * " profile not found in " * filepath)
             end
-        catch e
-            println("Demand flexibility parameters not found in " * filepath)
-            println(
-                "Demand flexibility parameters will default to allowing demand "
-                * "flexibility to occur."
-            )
-        end
 
-        # Check the feasibility of the duration parameter
-        if demand_flexibility["duration"] == nothing
-            demand_flexibility["duration"] = interval
-        elseif demand_flexibility["duration"] > interval
-            @warn (
-                "Demand flexibility durations greater than the interval length are "
-                * "set equal to the interval length."
-            )
-            demand_flexibility["duration"] = interval
-        end
-        
-        # Try reading the cost for up- and down-shifting loads
-        for cost_file_suffix in ("up", "dn")
-            try 
-                demand_flexibility["cost_$cost_file_suffix"] = CSV.File(
-                    joinpath(filepath, "demand_flexibility_cost_$cost_file_suffix.csv")
+            # Try loading the demand flexibility cost profiles
+            try
+                demand_flexibility["cost_" * s] = CSV.File(
+                    joinpath(filepath, "demand_flexibility_cost_" * s * ".csv")
                 ) |> DataFrames.DataFrame
-                println("...loading demand flexibility $cost_file_suffix-shift cost profiles")
+                println("...loading demand flexibility " * s * "-shift cost profiles")
             catch e
                 println(
-                    "Demand flexibility down-shift cost profiles not found in " * filepath
+                    "Demand flexibility " 
+                    * s 
+                    * "-shift cost profiles not found in " 
+                    * filepath
```

Comment:
In general the cost profiles are optional for demand flexibility to be considered. Probably, we could add another sentence here saying zero cost of elastic demand is considered?

Reply:
Sure, I can add an additional sentence

Reply:
Done.